src/main/java/com/google/sps/servlets/utils/CommentAnalysis.java

```diff
@@ -0,0 +1,73 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets.utils;
+
+import com.google.api.services.youtube.model.CommentThread;
+import com.google.api.services.youtube.model.CommentThreadListResponse;
+import com.google.cloud.language.v1.Document;
+import com.google.cloud.language.v1.LanguageServiceClient;
+import com.google.cloud.language.v1.Sentiment;
+import java.io.IOException;
+import java.util.ArrayList;
+
+
+/**
+ * This class encapsulates the each element in json comment array into separate user comment object
+ * and do sentiment analysis on each of them.
+ */
+public class CommentAnalysis {
+  private CommentThreadListResponse youtuberesponse;
+  private LanguageServiceClient languageService;
+
+  public CommentAnalysis(CommentThreadListResponse youtuberesponse) throws IOException {
+    this.youtuberesponse = youtuberesponse;
+    this.languageService = LanguageServiceClient.create();
```

Comment:
We should be careful having a `LanuageServiceClient` object that we could potentially not clean up.

From the [documentation](https://googleapis.dev/java/google-cloud-language/latest/com/google/cloud/language/v1/LanguageServiceClient.htm):
`Note: close() needs to be called on the languageServiceClient object to clean up resources such as threads. In the example above, try-with-resources is used, which automatically calls close().`

Functionally, not closing this client will result in resource-leaks on the backend and potentially increased memory usage. 

There are several approaches that you can take to deal with this problem. Three possibilities are:

1. Do not associate a `CommentThreadListResponse` with each instance of the `CommentAnalysis` object. and instead have one CommentAnalysis object that is shareable. In this way, we'll only have one `LanuageServiceClient` that is every initialized over the lifetime of the server. (This would mean changing the `computeOverallStats` to take a `CommentThreadListResponse`.

2. Create a new `LanguageServiceClient` every time that you want to perform analysis. The creation of new clients is relatively lightweight and should not be of too much concern.

3. Make this `CommentAnalysis` implement the [Autocloseable](https://docs.oracle.com/javase/8/docs/api/java/lang/AutoCloseable.html) interface.

Reply:
Update the code with the first proposal, but still kind of confused and asked in the conversation:)